src/PageObject/AdminPage.php

```diff
@@ -0,0 +1,148 @@
+<?php
+namespace PaulGibbs\WordpressBehatExtension\PageObject;
+
+use SensioLabs\Behat\PageObjectExtension\PageObject\Page;
+
+/**
+ * Page object representing a generic admin page.
+ *
+ * This class houses methods for navigating around the WordPress admin and
+ * acts as a base for most page-specific admin page objects.
+ */
+class AdminPage extends Page
+{
+    /**
+     * Returns the text in the header tag.
+     *
+     * The first h1 element is used, or first h2 element if it is not present.
+     *
+     * @return string The text in the header tag.
+     */
+    public function getHeaderText()
+    {
+        $header = $this->getHeaderElement();
+        $header_text = $header->getText();
+        $header_link = $header->find('css', 'a');
+
+        // The page headers can often incude an 'add new link'. Strip that out of the header text.
+        if ($header_link) {
+            $header_text  = trim(str_replace($header_link->getText(), '', $header_text));
+        }
+
+        return $header_text;
+    }
+
+    /**
+     * Asserts the text in the header tag matches the given text.
+     *
+     * @throws \Exception
+     */
+    public function assertHasHeader($expected)
+    {
+        $actual = $this->getHeaderText();
+        if ($expected !== $actual) {
+            throw new \Exception(sprintf('Expected page header "%s", found "%s".', $expected, $actual));
+        }
+    }
+
+    /**
+     * Get the text in the header tag/
+     *
+     * The first h1 element is used, or first h2 element if it is not present.
+     *
+     * @throws \Exception
+     */
+    private function getHeaderElement()
```

Comment:
Where do we stand on breaking changes? The advantage of `private` is that we know we can change or remove the function without breaking anything. If someone needs access to it, they can copy it into their own code. We're still at the alpha stage so it's not an issue - but if we want to avoid breaking changes going forward we might need to think critically about the API we are exposing.